modules/assim.sequential/R/hop_test.R

```diff
@@ -0,0 +1,98 @@
+##' @title hop_test
+##' @name  hop_test
+##' @author Ann Raiho \email{araiho@@nd.edu}
+##' 
+##' @param settings    SDA PEcAn settings object
+##'
+##' @description Hop test. This script tests that the model successfully reads it's own restart and can restart without loss of information.
+##' 
+##' @return NONE
+##' @export
+##' 
+hop_test <- function(settings, ens.runid = NULL){
+  ##### 
+  ##### Variable to check
+  ##### 
+  hop_var <- lapply(settings$state.data.assimilation$state.variables,'[[','variable.name')
+  
+  ##### 
+  ##### Regular Run
+  ##### 
+  if(is.null(ens.runid)){
+    run.write.configs(settings, write = settings$database$bety$write)
+    
+    PEcAn.remote::start.model.runs(settings, settings$database$bety$write)
+    
+    ens.runid <- read.table(file.path(settings$rundir,'runs.txt'))
+  }
+  ens <- read.output(runid = ens.runid, 
+                     outdir = file.path(settings$outdir,'out', ens.runid), 
+                     start.year = lubridate::year(settings$run$start.date), 
+                     end.year = lubridate::year(settings$run$end.date), 
+                     variables = hop_var)
+  ##### 
+  ##### Create Empty Data
+  ##### 
+  reg_run_end <- lubridate::year(settings$run$end.date)
+  settings$run$end.date <- paste0(reg_run_end - 9,'/12/31') #changing the run end.date so that the .nc files get written after write_restart
```

Comment:
why `9`?

Reply:
so that the hop test can be run for 10 years

Reply:
sorry, I meant to ask why 10 years? maybe this can be made a function argument, or at least it can be documented in the fcn description that the function runs for 10 years, for some models this can be too long

Reply:
I agree with @istfer that 10 years should not be hard coded (both here and below), but should depend on the start & end dates